### PR TITLE
[JENKINS-43802] Implementing retrieve/retrieveRevisions with an Item context

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/scm-api-plugin/pull/65.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <java.level>8</java.level>
     <no-test-jar>false</no-test-jar>
     <concurrency>1C</concurrency>
-    <scm-api-plugin.version>2.3.0</scm-api-plugin.version>
+    <scm-api-plugin.version>2.5.0-rc464.915933672136</scm-api-plugin.version> <!-- TODO https://github.com/jenkinsci/scm-api-plugin/pull/65 -->
```
Comment (bitwiseman) [inline pom.xml]:
```suggestion
        <scm-api-plugin.version>2.6.2</scm-api-plugin.version>
```

File: pom.xml [hunk 2]
```diff
@@ -31,7 +31,7 @@
     <no-test-jar>false</no-test-jar>
     <useBeta>true</useBeta>
     <concurrency>1C</concurrency>
-    <scm-api-plugin.version>2.3.0</scm-api-plugin.version>
+    <scm-api-plugin.version>2.6.2</scm-api-plugin.version>
```
Comment (bitwiseman) [inline pom.xml]:
```suggestion
    <scm-api-plugin.version>2.6.3</scm-api-plugin.version>
```

**Comment by bitwiseman** [comment]
@MarkEWaite 
What else is needed here to merge this?

**Comment by jglick** [comment]
> [@MarkEWaite](/MarkEWaite) [MarkEWaite](/MarkEWaite) added the [enhancement](/jenkinsci/git-plugin/labels/enhancement)  label [4 days ago](#event-2480542219)

I have been tracking analogous PRs as `bug`s, since from a user perspective it is.

**Comment by MarkEWaite** [comment]
> I have been tracking analogous PRs as `bug`s, since from a user perspective it is.

I've switched to call it a bug.

The label is used by release drafter to automate the placement of things in the draft release notes.  I'm fine with calling it a bug.  I've never gained any real benefit from distinguishing between the words "bug" and "feature".  I love the definition of "bug" that is attributed to James Bach and Michael Bolton.  They say “A bug is something that bugs somebody who matters.”

**Pull request merged**